### ISSUE TITLE
Support exact leaf values estimation on GPU for MAE, MAPE, Quantile

### ISSUE THREAD

**Issue by BeomyoungKim** [catboost/catboost#1313]
Problem: catboost gpu bad learning
catboost version: 0.23.2
Operating System: Linux 18.04
CPU: intel xeon silver 4114 2.2ghz 10c/20t
GPU: titan rtf

when using cpu, I got the pretty good scores for regression problem
but when I change to gpu, loss function(MAE) starts from 10000 and reduced very slowly when using default learning rate.
when using cpu, also starts from 10000 but reduced a lot per one Iter when using default learning rate.

around 1800 is learned loss when using cpu

**Comment by kizill** [comment]
Could you give us some code snippet so we can easily reproduce this issue?
You can try using some public datasets, share your own or generate some artificial dataset.

**Comment by brcharron** [comment]
@kizill I observed this behavior as well. [Here](https://colab.research.google.com/drive/1wl5oKEdWtz7eWd0MIrWLjQ0X3TKC3GNU?usp=sharing) is an example in Colab. With 1000 iterations, the CPU version reaches a RMSE of `19.3405866` after `2.21s` while the GPU version only `33.0687008` after `35.9s`.

```py
import numpy as np
from catboost.datasets import titanic
from catboost import CatBoostRegressor
train_df = titanic()[0][['Pclass', 'Age', 'SibSp', 'Parch', 'Embarked', 'Fare']].dropna()
X = train_df.drop(columns='Fare')
y = train_df.Fare
cat_features = np.where(X.dtypes != np.float)[0]
model_cpu = CatBoostRegressor(cat_features=cat_features, random_seed=0, task_type='CPU')
model_cpu.fit(X, y)
model_gpu = CatBoostRegressor(cat_features=cat_features, random_seed=0, task_type='GPU')
model_gpu.fit(X, y)
```

**Comment by RiverShah** [comment]
Linking these issues #1257

**Comment by LyzhinIvan** [comment]
Implemented in v0.25.
